rcl/src/rcl/event.c

```diff
@@ -190,6 +190,20 @@ rcl_event_get_rmw_handle(const rcl_event_t * event)
   }
 }
 
+bool
+rcl_event_is_valid(const rcl_event_t * event)
+{
+  RCL_CHECK_FOR_NULL_WITH_MSG(event, "event pointer is invalid", return false);
+  RCL_CHECK_FOR_NULL_WITH_MSG(event->impl, "event's implementation is invalid", return false);
+  /*
+  if (event->impl->rmw_handle.event_type == RMW_EVENT_INVALID) {
```

Comment:
I don't see much value of this if we don't check that the `rmw` event is also valid

Reply:
My initial proposal included that check, but I wasn't able to make it work. I think it requires making the typedef of the event impl exposed in another header. (Similar to the other `module_impl.h` headers). As this change required another change (exposing a typedef in a header) I wasn't sure it was desired to include two changes in one PR.

If you think it's not a problem I can make a PR with the whole changes for this to happen.

Reply:
> My initial proposal included that check, but I wasn't able to make it work. I think it requires making the typedef of the event impl exposed in another header

I'm not sure what typedef you're referring to.
The definition of the event implementation is defined in this same file https://github.com/ros2/rcl/blob/5d1962f3be6e9b8d6325af42ba9f0bd40b3db7ab/rcl/src/rcl/event.c#L36-L40.

> If you think it's not a problem I can make a PR with the whole changes for this to happen.

Yeah, I think it would be good to have a "final" version of the function on this PR.

Reply:
> The definition of the event implementation is defined in this same file

It is available there, so I can write in the `event.c` the code to check for `rmw_handle.implementation_identifier`. But I can't write the test it out because the type is not available externally. My suggestion for this is using a `event_impl.h` header as with other modules. I haven't tested it, but I think it will solve the issue.

Reply:
> My suggestion for this is using a event_impl.h header as with other modules. I haven't tested it, but I think it will solve the issue.

Ah yeah, it's not super nice to use the private definition in the tests but it sounds ok.

Reply:
Added this check, moved the definition to a private header and added the test scenario for it. Let me know if you think I should change the name of this PR accordingly.